networkit/cpp/io/RBGraphReader.cpp

```diff
@@ -0,0 +1,38 @@
+#include <networkit/auxiliary/Enforce.hpp>
+#include <networkit/auxiliary/Log.hpp>
+#include <networkit/auxiliary/StringTools.hpp>
+#include <networkit/io/RBGraphReader.hpp>
+#include <networkit/io/RBMatrixReader.hpp>
+
+namespace NetworKit {
+
+Graph RBGraphReader::read(std::string_view path) {
+    std::ifstream in(path.data());
+    if (!in.is_open()) {
+        throw std::runtime_error("could not open: " + std::string(path));
+    }
+
+    RBMatrixReader reader;
+
+    reader.readToVectors(in);
```

Comment:
Hm, we effectively allocate memory for edges twice. How about implementing the logic from `readToVectors` in a parser class and either then use it to fill the vectors or the graph directly?